### PR TITLE
memory: Initial support for checking memory

### PR DESCRIPTION
**Pull request by stevenh**
Initial support for:
* Reading memory data
* Resetting memory counters
* Setting memory scrub rate

### CODE REVIEW

File: lib/edac/memory_controller.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	if err != nil {
+		return err
+	}
+
+	*addr = v
+
+	return nil
+}
+
+// loadString loads a string from file into.
```
Comment (darkliquid) [inline lib/edac/memory_controller.go]: into what?

File: lib/edac/memory_types.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+package edac
+
+import (
+	"fmt"
+	"time"
+)
+
+// Information sourced from:
+// https://github.com/torvalds/linux/blob/master/Documentation/ABI/testing/sysfs-devices-edac
+
```
Comment (darkliquid) [inline lib/edac/memory_types.go:10]: Missing golint compat comment here

File: lib/edac/memory_types.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+func (mi MemoryInfo) HasErrors() bool {
+	switch {
+	case mi.Correctable != 0:
+		return true
+	case mi.CorrectableNoInfo != 0:
+		return true
+	case mi.Uncorrectable != 0:
+		return true
+	case mi.UncorrectableNoInfo != 0:
+		return true
```
Comment (darkliquid) [inline lib/edac/memory_types.go:76]: not sure if doing `return mi.Correctable+mi.CorrectableNoInfo+mi.Uncorrectable+mi.UncorrectableNoInfo > 0` is a better solution here or not

Comment (darkliquid) [reply lib/edac/memory_types.go:76]: Not sure if these values can ever be negative or not

Comment (lwaddicor) [reply lib/edac/memory_types.go:76]: The documentation for some of these fields states "This count field should be monitored for non-zero values and report such information to the system administrator." This gives no guarantee that these are not negative

Comment (lwaddicor) [reply lib/edac/memory_types.go:76]: https://www.kernel.org/doc/html/v4.12/admin-guide/ras.html

Comment (darkliquid) [reply lib/edac/memory_types.go:76]: hmm, yeah, I'll revert that back to the previous switch behaviour then

Comment (stevenh) [reply lib/edac/memory_types.go:76]: Updated to fix Stringer comments, there was two and moved the sourced comment as that was for all types not just MemoryInfo.

**Review by darkliquid — APPROVED** [review]

**Pull request merged**